client/components/contact/ContactForm.jsx

```diff
@@ -31,13 +30,13 @@ const initialFormValues = {
 };
 
 const ContactForm = () => {
-  // define the methods to dispatch redux actions
+  // Define the methods to dispatch redux actions.
   const dispatch = useDispatch();
   const callSendGitRequest = useCallback(obj => dispatch(sendGitRequest(obj)), [dispatch]);
```

Comment:
I'm learning more about these hooks, and I'm wondering if it's necessary (or potentially harmful) to use `useCallback` here. It seems like `useCallback` is only useful if we're passing a function to a child component and that child component is trying to optimize render behavior in some way. Wdyt?

[redux useDispatch docs](https://react-redux.js.org/api/hooks#usedispatch)

Reply:
Yes, my understanding is useCallback just caches the function so it doesn't have to be recreated at render if the dependencies in the dependencies array do not change. However, I'm getting the following eslint warnings:

```
/home/ed/311-data/client/components/contact/ContactForm.jsx
  35:9  warning  The 'callSendGitRequest' function makes the dependencies of useCallback Hook (at line 146) change on every render. Move it inside the useCallback callback. Alternatively, wrap the 'callSendGitRequest' definition into its own useCallback() Hook       react-hooks/exhaustive-deps
  36:9  warning  The 'callShowFeedbackSuccess' function makes the dependencies of useEffect Hook (at line 69) change on every render. Move it inside the useEffect callback. Alternatively, wrap the 'callShowFeedbackSuccess' definition into its own useCallback() Hook  react-hooks/exhaustive-deps
  37:9  warning  The 'callShowErrorModal' function makes the dependencies of useEffect Hook (at line 69) change on every render. Move it inside the useEffect callback. Alternatively, wrap the 'callShowErrorModal' definition into its own useCallback() Hook            react-hooks/exhaustive-deps

```
They are not technically errors so I can technically still push changes up. So, would you like me to ignore them and push up without use of useCallback(), add some eslint ignore to mute the warnings or keep as is? Thanks

Reply:
getting rid of the lambdas and making the dispatch calls directly where needed as per discussion

Reply:
updated in latest push